src/posix/platform/netif.cpp

```diff
@@ -379,6 +793,284 @@ static void processNetifLinkEvent(otInstance *aInstance, struct nlmsghdr *aNetli
+                        ifr6.ifra_addr.sin6_family = AF_INET6;
+                        ifr6.ifra_addr.sin6_len    = sizeof(ifr6.ifra_addr);
+                        memcpy(&ifr6.ifra_addr.sin6_addr, &addr6.sin6_addr, sizeof(struct in6_addr));
+                        ifr6.ifra_prefixmask.sin6_family = AF_INET6;
+                        ifr6.ifra_prefixmask.sin6_len    = sizeof(ifr6.ifra_prefixmask);
+                        memset(&ifr6.ifra_prefixmask.sin6_addr, 0xFF, 16);
+                        apply_mask(&ifr6.ifra_prefixmask.sin6_addr, netAddr.mPrefixLength);
+                        ifr6.ifra_lifetime.ia6t_vltime = ND6_INFINITE_LIFETIME;
+                        ifr6.ifra_lifetime.ia6t_pltime = ND6_INFINITE_LIFETIME;
+
+#if defined(__APPLE__)
+                        ifr6.ifra_lifetime.ia6t_expire    = ND6_INFINITE_LIFETIME;
+                        ifr6.ifra_lifetime.ia6t_preferred = ND6_INFINITE_LIFETIME;
+
+                        ifr6.ifra_flags |= IN6_IFF_NODAD;
+#endif
+
+                        err = ioctl(sIpFd, SIOCDIFADDR_IN6, &ifr6);
+                        if (err != 0)
+                        {
+                            otLogWarnPlat(
+                                "error (%d) removing stack-addded link-local address %s", errno,
+                                inet_ntop(AF_INET6, addr6.sin6_addr.s6_addr, addressString, sizeof(addressString)));
+                        }
+                        else
+                        {
+                            otLogNotePlat(
+                                "        %s (removed stack-added link-local)",
+                                inet_ntop(AF_INET6, addr6.sin6_addr.s6_addr, addressString, sizeof(addressString)));
+                        }
+                    }
+                    else
+                    {
+                        // we don't know about this address -- we need to add it
+                        error = otIp6AddUnicastAddress(aInstance, &netAddr);
+
+                        if ((error == OT_ERROR_NONE) || (error == OT_ERROR_ALREADY))
+                        {
+                            otLogNotePlat(
+                                "ADD [U] %s%s",
+                                inet_ntop(AF_INET6, addr6.sin6_addr.s6_addr, addressString, sizeof(addressString)),
+                                error == OT_ERROR_ALREADY ? " (already subscribed, ignored)" : "");
+                        }
+                        else
+                        {
+                            otLogWarnPlat(
+                                "ADD [U] %s failed (%s)",
+                                inet_ntop(AF_INET6, addr6.sin6_addr.s6_addr, addressString, sizeof(addressString)),
+                                otThreadErrorToString(error));
+                        }
+                    }
+                }
+            }
+            else
+            {
+                otNetifMulticastAddress netAddr;
```

Comment:
A new function as well?

Reply:
I consolidated a lot of the logging that was making this so complicated.  Take a peek.